WikiDrips/WikiTableViewController.swift

```diff
@@ -94,17 +93,31 @@ class WikiTableViewController: UITableViewController {
                 if newDocs.count > 0 {
                     strongSelf.clearPendingImageTasks() // Remove stale tasks created by consecutive searches
                     strongSelf.pendingSearch = false
-                    if isFirstResult {
-                        strongSelf.wikiDocs = [newDocs]
-                    } else {
-                        strongSelf.wikiDocs[0] += (newDocs)
-                    }
-                    strongSelf.tableView.reloadData()
+                    strongSelf.populateRows(atOffset: wikiRequest.offset, with: newDocs)
+                    
                 }
             }
         }
     }
     
+    fileprivate func populateRows(atOffset offset: Int, with newDocs: [WikiDoc]) {
+        if offset == 0 {
+            wikiDocs = [newDocs]
+            tableView.reloadData()
+        } else {
+            let oldCount = wikiDocs[0].count
+            tableView.beginUpdates()
+            wikiDocs[0] += (newDocs)
+            let newDocCount = wikiDocs[0].count - 1
+            var rows = [IndexPath]()
+            for index in oldCount...newDocCount {
```

Comment:
Not a big deal, but instead of subtracting 1 from newDocCount (and thus not really being the count) you could use the `..<` operator:
```
            let newDocCount = wikiDocs[0].count
            var rows = [IndexPath]()
            for index in oldCount..<newDocCount {
```